pkg/agent/util/syscall/syscall_windows.go

```diff
@@ -0,0 +1,397 @@
+		prefixLength: uint8(prefixLength),
+	}
+}
+
+// NlRouteProtocol defines the routing mechanism that an IP route was added with.
+type NlRouteProtocol uint32
+
+const (
+	RouteProtocolOther   NlRouteProtocol = 1
+	RouteProtocolLocal   NlRouteProtocol = 2
+	RouteProtocolNetMgmt NlRouteProtocol = 3
+	RouteProtocolIcmp    NlRouteProtocol = 4
+	RouteProtocolEgp     NlRouteProtocol = 5
+	RouteProtocolGgp     NlRouteProtocol = 6
+	RouteProtocolHello   NlRouteProtocol = 7
+	RouteProtocolRip     NlRouteProtocol = 8
+	RouteProtocolIsIs    NlRouteProtocol = 9
+	RouteProtocolEsIs    NlRouteProtocol = 10
+	RouteProtocolCisco   NlRouteProtocol = 11
+	RouteProtocolBbn     NlRouteProtocol = 12
+	RouteProtocolOspf    NlRouteProtocol = 13
+	RouteProtocolBgp     NlRouteProtocol = 14
+	RouteProtocolIdpr    NlRouteProtocol = 15
+	RouteProtocolEigrp   NlRouteProtocol = 16
+	RouteProtocolDvmrp   NlRouteProtocol = 17
+	RouteProtocolRpl     NlRouteProtocol = 18
+	RouteProtocolDhcp    NlRouteProtocol = 19
+
+	//
+	// Windows-specific definitions.
+	//
+	NT_AUTOSTATIC     NlRouteProtocol = 10002
+	NT_STATIC         NlRouteProtocol = 10006
+	NT_STATIC_NON_DOD NlRouteProtocol = 10007
+)
+
+// NlRouteOrigin defines the origin of the IP route.
+type NlRouteOrigin uint32
+
+const (
+	NlroManual              NlRouteOrigin = 0
+	NlroWellKnown           NlRouteOrigin = 1
+	NlroDHCP                NlRouteOrigin = 2
+	NlroRouterAdvertisement NlRouteOrigin = 3
+	Nlro6to4                NlRouteOrigin = 4
+)
+
+type MibIPForwardRow struct {
+	Luid              uint64
+	Index             uint32
+	DestinationPrefix AddressPrefix
+	NextHop           RawSockAddrInet
+
+	SitePrefixLength  uint8
+	ValidLifetime     uint32
+	PreferredLifetime uint32
+	Metric            uint32
+	Protocol          NlRouteProtocol
+
+	Loopback             bool
+	AutoconfigureAddress bool
+	Publish              bool
+	Immortal             bool
+
+	Age    uint32
+	Origin NlRouteOrigin
+}
+
+type MibIPForwardTable struct {
+	NumEntries uint32
+	Table      [1]MibIPForwardRow
+}
+
+var (
+	modiphlpapi = syscall.NewLazyDLL("iphlpapi.dll")
+
+	procGetIPInterfaceEntry  = modiphlpapi.NewProc("GetIpInterfaceEntry")
+	procSetIPInterfaceEntry  = modiphlpapi.NewProc("SetIpInterfaceEntry")
+	procCreateIPForwardEntry = modiphlpapi.NewProc("CreateIpForwardEntry2")
+	procDeleteIPForwardEntry = modiphlpapi.NewProc("DeleteIpForwardEntry2")
+	procGetIPForwardTable    = modiphlpapi.NewProc("GetIpForwardTable2")
+	procFreeMibTable         = modiphlpapi.NewProc("FreeMibTable")
+)
+
+type NetIOInterface interface {
+	GetIPInterfaceEntry(ipInterfaceRow *MibIPInterfaceRow) (errcode error)
+
+	SetIPInterfaceEntry(ipInterfaceRow *MibIPInterfaceRow) (errcode error)
+
+	CreateIPForwardEntry(ipForwardEntry *MibIPForwardRow) (errcode error)
+
+	DeleteIPForwardEntry(ipForwardEntry *MibIPForwardRow) (errcode error)
+
+	freeMibTable(table unsafe.Pointer)
+
+	getIPForwardTable(family uint16, ipForwardTable **MibIPForwardTable) (errcode error)
```

Comment:
No need to have these methods in the interface if they are not supposed to be used externally.